### ISSUE TITLE
Deprecating `fastify-*` modules

### ISSUE THREAD

**Issue by jsumners** [fastify/fastify#3733]
2022-05-07 remaining work:

The following list is hoisted from comment https://github.com/fastify/fastify/issues/3733#issuecomment-1114252976

- [x] [aws-lambda-fastify](https://github.com/fastify/aws-lambda-fastify)
  - Rename package (https://github.com/fastify/aws-lambda-fastify/pull/110)
- [x] [fastify-auth](https://github.com/fastify/fastify-auth)
  - Update dependencies (https://github.com/fastify/fastify-auth/pull/154)
- [x] [fastify-awilix](https://github.com/fastify/fastify-awilix)
  - Update dependencies (https://github.com/fastify/fastify-awilix/pull/73)
- [x] [fastify-csrf](https://github.com/fastify/fastify-csrf)
  - Rename package
- [x] [fastify-flash](https://github.com/fastify/fastify-flash)
  - Update dependencies
- [x] [fastify-schedule](https://github.com/fastify/fastify-schedule)
  - Update dependencies (https://github.com/fastify/fastify-schedule/pull/82/)
- [x] [fastify-session](https://github.com/fastify/session)
  - Missing release notes
  - Update dependencies
- [x] [fastify-swagger](https://github.com/fastify/fastify-swagger)
  - Update dependencies
- [x] [middie](https://github.com/fastify/middie)
  - Rename package (https://github.com/fastify/middie/pull/148)
- [x] [point-of-view](https://github.com/fastify/point-of-view)
  - Rename package (https://github.com/fastify/point-of-view/pull/325)
- [x] [under-pressure](https://github.com/fastify/under-pressure)
  - Rename package (https://github.com/fastify/under-pressure/pull/166)
- [x] [any-schema-you-like](https://github.com/fastify/any-schema-you-like)
  - Rename package (https://github.com/fastify/any-schema-you-like/pull/25)

----
Original post:

### Issue

If we are going to deprecate `fastify-*` modules, as outlined in https://github.com/fastify/fastify/issues/3482#issuecomment-1046916608, then I really need help with https://github.com/fastify/deprecate-modules.

To start, I need help auditing https://github.com/fastify/deprecate-modules/blob/a7a87b05d129e04d7e36bd4f0eb07d479e2d7b6f/lib/modules.json to determine:

1. Is the list of modules correct
2. Are the versions correct? `versionToPublish` will be the new `fastify-*` version for the module. The `newModule.version` will be the version published under `@fastify/*`.

Subsequent to that, I need help with the todo list in the readme. In particular, is there anything that needs to be done for TypeScript in the deprecated modules (look at the templates)? How should the templates be updated for ESM exports?

Once we have the deprecation portions written and tested, we need to work on adding script for:

1. Updating GitHub repository information (rename the repo, update links in package.json, etc)
2. Patching the `package.json` with the new name and version for the new `@fastify/*` packages

attn: @fastify/core

**Comment by zekth** [comment]
Could we make a checklist for each package like:

**fastify-cors:**
- [ ] Fastify 4 compliant
- [ ] Upgrade version OK
- [ ] Template compliant. (doc / typescript )

**Comment by zekth** [comment]
Project has been created: https://github.com/fastify/deprecate-modules/projects/1

**Comment by mcollina** [comment]
@zekth we will be changing name and bumping major and then migrate to fastify v4 and bump major again.

**Comment by zekth** [comment]
> @zekth we will be changing name and bumping major and then migrate to fastify v4 and bump major again.

Right now we're doing straight deprecation and work with the compatibility and compliancy afterward so?

**Comment by mcollina** [comment]
yes exactly

**Comment by jsumners** [comment]
As [currently written](https://github.com/fastify/deprecate-modules/blob/a7a87b05d129e04d7e36bd4f0eb07d479e2d7b6f/lib/modules.json#L119-L127):

```json
  {
    "name": "fastify-cookie",
    "versionToPublish": "5.7.0",
    "license": "MIT",
    "newModule": {
      "name": "@fastify/cookie",
      "version": "6.0.0"
    }
  },
```

The following would happen:

1. A new `fastify-cookie` package would be published at version 5.7.0 (up from 5.6.0). This package will wrap the exports of 5.6.0 with a deprecation notice.
2. A new `@fastify/cookie` package would be published at version 6.0.0. This package will have the same code as 5.6.0.

Any subsequent changes to comply with `fastify@4` would bump `@fastify/cookie` to 7.0.0.

This is the correct plan, yes?

**Comment by mcollina** [comment]
amazing!

**Comment by yaneony** [comment]
What about "point-of-view" plugin? Maybe it makes sense to rename it as **@fastify/views** ?

**Comment by jsumners** [comment]
> What about "point-of-view" plugin? Maybe it makes sense to rename it as **@fastify/views** ?

This one _does_ get skipped by the automated tool. I don't really have an opinion about moving it under the org scope. It's already named independently, and thus does not cause confusion around whether or not it is a community plugin or an org plugin.

**Comment by yaneony** [comment]
But isn't "point-of-view" one of org plugins?

**Comment by jsumners** [comment]
> But isn't "point-of-view" one of org plugins?

Yes.

**Comment by yaneony** [comment]
But then, it makes sense to move it to org plugins as well. As of mine sight of view. Right now this plugin looks "out of scope" and not fastify related. Moving it to @fastify/views or @fastify/pov will make it look more fastify related.

Just only my 2 cents :D

**Comment by fox1t** [comment]
I like this plan! 👏

Speaking of `point-of-view` I vote for `@fastify/point-of-view` and it can follow the same upgrade path as others.

**Comment by mcollina** [comment]
I would just move it to `@fastify/view`

**Comment by jsumners** [comment]
Given that `point-of-view` does not conform to the other modules's naming style, it would be difficult to include in the automated tool. It should be done manually.

**Comment by delvedor** [comment]
I would keep the same name to avoid too much confusion, so `@fastify/point-of-view`.
Otherwise, `@fastify/view` works as well.

**Comment by mcollina** [comment]
@jsumners is on it, hopefully shipping this in the next two weeks!

**Comment by jsumners** [comment]
My next step is to do a live test with `fastify-bearer-auth` -- https://github.com/fastify/deprecate-modules/commit/c9676b0413829f5c4d182e8b539d15e9cb918120

I'll probably do that tomorrow. If that is successful, we will need to coordinate a freeze time wherein I can issue all of the deprecations and generate the PRs. I am aiming for this to be early next week (last week of April 2022).

**To be clear:**

All modules in the org with names that start with `fastify-` will:

1. Have a new current major version published that deprecates said major version.
2. Be renamed to `@fastify/<module_name>` via a pull request to the module's repo.

There a couple consequences:

1. No further releases of the _current major_ can ever be released. The code for deprecating the module is ephemeral.
2. We will need to merge all PRs and issue new releases manually.

attn: @mcollina @fastify/core

**Comment by jsumners** [comment]
The deprecation test is working with commit https://github.com/fastify/deprecate-modules/commit/73f6805c0681e849c250cde4f6e5b701d58d2086. The below test shows the result of the test.

Unfortunately, I'm blocked on testing the new scoped package due to https://twitter.com/jsumners79/status/1517138149817815040

That thread also shows that we are blocked by the configuration on all of the npm packages requiring 2FA instead of "2FA or automation token".

## Deprecation Test

<details>
  <summary>package.json</summary>

```json
{
  "dependencies": {
    "fastify": "^3.28.0",
    "fastify-bearer-auth": "^6.3.0-rc.2"
  }
}
```

</details>

<details>
  <summary>index.js</summary>

```js
'use strict'

const fastify = require('fastify')({ logger: { level: 'debug' } })

fastify.register(
  require('fastify-bearer-auth'),
  { keys: ['abc123'] }
)

fastify.route({
  method: 'GET',
  path: '/',
  handler (req, res) {
    res.send({ hello: 'world' })
  }
})

fastify.listen(8000)

```
</details>

<details>
  <summary>Output</summary>

```
❯ node index.js
(node:23712) [FST_MODULE_DEP_FASTIFY-BEARER-AUTH] FastifyWarning.fastify-bearer-auth: fastify-bearer-auth has been deprecated. Use @fastify/bearer-auth@7.0.0-rc.1 instead.
(Use `node --trace-warnings ...` to show where the warning was created)
{"level":30,"time":1650548265416,"pid":23712,"hostname":"Jamess-MBP","msg":"Server listening at http://127.0.0.1:8000"}
```
</details>

**Comment by jsumners** [comment]
The non-deprecation test works as desired.

## Non-deprecation Test

<details>
  <summary>package.json</summary>

```json
{
  "dependencies": {
    "fastify": "^3.28.0",
    "@fastify/bearer-auth": "^7.0.0-rc.1"
  }
}
```

</details>

<details>
  <summary>index.js</summary>

```js
'use strict'

const fastify = require('fastify')({ logger: { level: 'debug' } })

fastify.register(
  require('@fastify/bearer-auth'),
  { keys: ['abc123'] }
)

fastify.route({
  method: 'GET',
  path: '/',
  handler (req, res) {
    res.send({ hello: 'world' })
  }
})

fastify.listen(8000)

```
</details>

<details>
  <summary>Output</summary>

```
❯ node index.js
{"level":30,"time":1650576011602,"pid":29251,"hostname":"Jamess-MacBook-Pro.local","msg":"Server listening at http://[::1]:8000"}
```
</details>

**Comment by climba03003** [comment]
Good to know that everything is working. I have left a comment in https://github.com/fastify/deprecate-modules/issues/1

**Comment by jsumners** [comment]
Before I can continue, we need to solve access to several modules (https://github.com/fastify/deprecate-modules/blob/5190b2a8df25dff4e7fd3e69bef555b708a9bcc7/build-data.mjs#L19-L35):

```js
const skipRepos = [
  'fastify-api', // not in Fastify npm org
  'fastify-awilix', // not in Fastify npm org
  'fastify-citgm', // not an npm package
  'fastify-csrf', // under Matteo's npm account
  'fastify-diagnostics-channel', // not in Fastify npm org
  'fastify-dx', // not an npm package
  'fastify-error', // not in Fastify npm org
  'fastify-plugin', // should remain in general npm repo
  'fastify-routes-stats', // under Matteo's npm account
  'fastify-schedule', // under kibertoad's npm account
  'fastify-snippet', // not an npm package
  'fastify-starter-codesandbox', // not an npm package
  'fastify-test', // not an npm package
  'fastify-typescript-extended-sample', // not an npm package
  'fastify-vite', // not an npm package
]
```

Some of those should not be included in the process, but many should be. Basically, the following modules need to be added to the npm org or at least @mcollina and I need to sort out what the permissions should be:

+ fastify-api // not in Fastify npm org
+ fastify-awilix // not in Fastify npm org
+ fastify-csrf // under Matteo's npm account
+ fastify-diagnostics-channel // not in Fastify npm org
+ fastify-error // not in Fastify npm org
+ fastify-routes-stats // under Matteo's npm account
+ fastify-schedule // under kibertoad's npm account


attn: @mcollina @kibertoad @galvez

**Comment by RafaelGSS** [comment]
I'll move the `fastify-diagnostics-channel` to Fastify npm org

**Comment by mcollina** [comment]
I'll do my best to move them.

**Comment by mcollina** [comment]
I've fixed all the permissions for the modules I have rights to.

Unfortunately we have https://github.com/fastify/csrf that is published to `@fastify/csrf`, so we need a different target for `fastify-csrf` (or possibly manually handle it).

**Comment by jsumners** [comment]
The only modules left are those started by @kibertoad and the `fastify-api` module by @galvez. If they are unable to assist by the morning (EDT) of 2022-04-27 then I'll proceed and those modules can be handled later.

**Comment by Fdawgs** [comment]
I'll spin up a script to open PRs to re-point things like readme npm badges once this process is complete.

**Comment by jsumners** [comment]
> I'll spin up a script to open PRs to re-point things like readme npm badges once this process is complete. Looking good so far!
> 
> References to repos, like those in https://github.com/fastify/fastify/blob/main/docs/Guides/Ecosystem.md, will also need updating.

Please allow me to write up a summary after the fact. It will be posted here when I have reached completion.

**Comment by jsumners** [comment]
For reference, this is the build log.

<details>
  <summary>build.log.json</summary>

```json
{
  "fastify-accepts": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-accepts/pull/128"
    }
  },
  "fastify-accepts-serializer": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-accepts-serializer/pull/74"
    }
  },
  "fastify-auth": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-auth/pull/147"
    }
  },
  "fastify-autoload": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.13.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-autoload/pull/233"
    }
  },
  "fastify-basic-auth": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-basic-auth/pull/84"
    }
  },
  "fastify-bearer-auth": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "6.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-bearer-auth/pull/130"
    }
  },
  "fastify-caching": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "6.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-caching/pull/99"
    }
  },
  "fastify-circuit-breaker": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-circuit-breaker/pull/86"
    }
  },
  "fastify-compress": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-compress/pull/221"
    }
  },
  "fastify-cookie": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "5.7.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-cookie/pull/177"
    }
  },
  "fastify-cors": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "6.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-cors/pull/195"
    }
  },
  "fastify-diagnostics-channel": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-diagnostics-channel/pull/37"
    }
  },
  "fastify-elasticsearch": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-elasticsearch/pull/74"
    }
  },
  "fastify-env": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-env/pull/139"
    }
  },
  "fastify-error": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-error/pull/72"
    }
  },
  "fastify-etag": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-etag/pull/79"
    }
  },
  "fastify-express": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.4.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-express/pull/78"
    }
  },
  "fastify-flash": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-flash/pull/66"
    }
  },
  "fastify-formbody": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "5.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-formbody/pull/107"
    }
  },
  "fastify-funky": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "1.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-funky/pull/95"
    }
  },
  "fastify-helmet": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "7.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-helmet/pull/177"
    }
  },
  "fastify-hotwire": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-hotwire/pull/53"
    }
  },
  "fastify-http-proxy": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "6.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-http-proxy/pull/238"
    }
  },
  "fastify-jwt": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-jwt/pull/220"
    }
  },
  "fastify-kafka": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-kafka/pull/66"
    }
  },
  "fastify-leveldb": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-leveldb/pull/91"
    }
  },
  "fastify-mongodb": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-mongodb/pull/177"
    }
  },
  "fastify-multipart": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "5.4.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-multipart/pull/340"
    }
  },
  "fastify-mysql": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-mysql/pull/90"
    }
  },
  "fastify-nextjs": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "7.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-nextjs/pull/522"
    }
  },
  "fastify-oauth2": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.6.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-oauth2/pull/147"
    }
  },
  "fastify-passport": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.5.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-passport/pull/532"
    }
  },
  "fastify-postgres": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.7.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-postgres/pull/128"
    }
  },
  "fastify-rate-limit": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "5.9.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-rate-limit/pull/236"
    }
  },
  "fastify-redis": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.4.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-redis/pull/143"
    }
  },
  "fastify-reply-from": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "6.7.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-reply-from/pull/244"
    }
  },
  "fastify-request-context": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-request-context/pull/110"
    }
  },
  "fastify-response-validation": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-response-validation/pull/67"
    }
  },
  "fastify-routes": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-routes/pull/62"
    }
  },
  "fastify-routes-stats": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-routes-stats/pull/71"
    }
  },
  "fastify-secure-session": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-secure-session/pull/122"
    }
  },
  "fastify-sensible": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-sensible/pull/105"
    }
  },
  "fastify-soap-client": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "0.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-soap-client/pull/44"
    }
  },
  "fastify-static": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.7.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-static/pull/286"
    }
  },
  "fastify-swagger": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "5.2.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-swagger/pull/586"
    }
  },
  "fastify-url-data": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "3.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-url-data/pull/101"
    }
  },
  "fastify-websocket": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "4.3.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-websocket/pull/194"
    }
  },
  "fastify-zipkin": {
    "deprecationModule": {
      "published": true,
      "versionPublished": "2.1.0"
    },
    "replacementModule": {
      "prCreated": true,
      "prUrl": "https://github.com/fastify/fastify-zipkin/pull/68"
    }
  }
}
```
</details>

**Comment by climba03003** [comment]
Follow Up PR
[`fastify-http-proxy`](https://github.com/fastify/fastify-http-proxy/pull/241) - fix the typescript error
[`fastify-passport`](https://github.com/fastify/fastify-passport/pull/534) - update some of the packages and docs

**Comment by yaneony** [comment]
2 questions

1. When will it be available on npmjs.
2. Will we need to reinstall modules or it will go with npm upgrade?

**Comment by jsumners** [comment]
Now. The names a different.

**Comment by jessekrubin** [comment]
will `fastify` stay `fastify` or become something like `@fastify/fastify` / `@fastify/core`?

**Comment by jsumners** [comment]
Please read above.

**Comment by jessekrubin** [comment]
So no.

**Comment by Fdawgs** [comment]
I think there may need to be a note stating that the plugin names have not changed, i.e. https://github.com/fastify/fastify-accepts/blob/master/index.js#L61

**Comment by yaneony** [comment]
Will they be changed? Because README file still has old names, see: https://github.com/fastify/fastify-accepts/blob/master/README.md#install

**Comment by Fdawgs** [comment]
> Will they be changed? Because README file still has old names, see: https://github.com/fastify/fastify-accepts/blob/master/README.md#install

See the Work To Be Done section in https://github.com/fastify/fastify/issues/3733#issuecomment-1111188683

**Comment by jsumners** [comment]
> I think there may need to be a note stating that the plugin names have not changed, i.e. https://github.com/fastify/fastify-accepts/blob/master/index.js#L61, to avoid confusion.

I think they should be updated to reference `pkg.name` from `package.json`. This work can target the next major version that will coincide with `fastify@4`.

**Comment by Fdawgs** [comment]
> ## Work To Be Done
> 
>     1. None of the READMEs were updated. There are several README files that reference the names of modules, other modules and their own self. These names should be updated by the community as they are able to work on it.
> 
>     2. The core list of modules needs to be updated -- https://github.com/fastify/fastify/blob/da8fafdc2132cde357230fa5b43344058fdd3586/docs/Guides/Ecosystem.md#ecosystem
> 
>     3. Repositories should probably renamed, e.g. `github.com/fastify/fastify-cookie` should probably become `github.com/fastify/cookie`. The associated `package.json` references should not be forgotten.
> 
>     4. Maybe some other stuff that I am overlooking.
> 

I've tackled 1, 2, and the `package.json` part of 3.

**Comment by jsumners** [comment]
Good work @Fdawgs. As for the `package.json` updates I was referring to: I meant the repository and homepage URLs. Those will need to change _if_ we decide to rename the repositories.

**Comment by Fdawgs** [comment]
> Good work @Fdawgs. As for the `package.json` updates I was referring to: I meant the repository and homepage URLs. Those will need to change _if_ we decide to rename the repositories.

Ah, okay, I thought you meant dependencies, I retract my confirmation of completing step 3. 😄

**Comment by jsumners** [comment]
@Fdawgs have you published new versions of the modules where you updated dependencies?

**Comment by Fdawgs** [comment]
> @Fdawgs have you published new versions of the modules where you updated dependencies?

Not something I have permissions for afaik, I can generate provide a list for them all if someone else would be so kind as to do it?

**Comment by jsumners** [comment]
Provide the list. Plugins team should be able to publish plugins if I'm not mistaken. We can sort that out.

**Comment by Fdawgs** [comment]
> Provide the list. Plugins team should be able to publish plugins if I'm not mistaken. We can sort that out.

Dev dependency updates:
- [fastify-autoload](https://github.com/fastify/fastify-autoload)
- [fastify-hotwire](https://github.com/fastify/fastify-hotwire)
- [fastify-jwt](https://github.com/fastify/fastify-jwt)
- [fastify-reply-from](https://github.com/fastify/fastify-reply-from)
- [fastify-static](https://github.com/fastify/fastify-static)
- [fastify-bearer-auth](https://github.com/fastify/fastify-bearer-auth)
- [fastify-basic-auth](https://github.com/fastify/fastify-basic-auth)

Prod dependency updates:
- [fastify-accepts-serializer](https://github.com/fastify/fastify-accepts-serializer)
- [fastify-swagger](https://github.com/fastify/fastify-swagger)
- [under-pressure](https://github.com/fastify/under-pressure)

**Comment by RafaelGSS** [comment]
Just checked all the plugins in our ecossystem and those ones is missing updates:

- [ ] [aws-lambda-fastify](https://github.com/fastify/aws-lambda-fastify)
  - Rename package
- [ ] [fastify-auth](https://github.com/fastify/fastify-auth)
  - Update dependencies
- [ ] [fastify-awilix](https://github.com/fastify/fastify-awilix)
  - Rename package
- [ ] [fastify-csrf](https://github.com/fastify/fastify-csrf)
  - Rename package
- [ ] [fastify-flash](https://github.com/fastify/fastify-flash)
  - Update dependencies
- [ ] [fastify-schedule](https://github.com/fastify/fastify-schedule)
  - Rename package
- [ ] [fastify-session](https://github.com/fastify/session)
  - Missing release notes
  - Update dependencies
- [ ] [fastify-swagger](https://github.com/fastify/fastify-swagger)
  - Update dependencies
- [ ] [middie](https://github.com/fastify/middie)
  - Rename package
- [ ] [point-of-view](https://github.com/fastify/point-of-view)
  - Rename package
- [ ] [under-pressure](https://github.com/fastify/under-pressure)
  - Rename package

Feel free to mark-as-complete when a package is done.

**Comment by jsumners** [comment]
@RafaelGSS those will need to be handled manually by someone with time. I am out of it.

**Comment by jsumners** [comment]
Here's another that needs attention: https://github.com/fastify/any-schema-you-like

**Comment by RafaelGSS** [comment]
> @RafaelGSS those will need to be handled manually by someone with time. I am out of it.

Fair enough. I have added to this issue just to keep it tracking, I'll probably work on some of them.

**Comment by jessekrubin** [comment]
Will the plugin names as they are given to `fastify-plugin` also change? 

Here is the newly renamed `@fastify/static` which gives fastify-plugin the name `fastify-static`: https://github.com/fastify/fastify-static/blob/dee5d8014c0831dc4721f3b09062a75d7348eead/index.js#L511

Should all of those names be changed? or no? (no where on this issue thread to I see a mention of this...)

**Comment by RafaelGSS** [comment]
See https://github.com/fastify/fastify/issues/3856 @jessekrubin

**Comment by jessekrubin** [comment]
@RafaelGSS the issue/thread you pointed at does not (at least to me) address what I am talking about. When you make a fastify plugin you give a plugin name (which probably know); will the names that are given when calling fp/FastifyPlugin be changed? Am I missing something?

**Comment by climba03003** [comment]
> @RafaelGSS the issue/thread you pointed at does not (at least to me) address what I am talking about. When you make a fastify plugin you give a plugin name (which probably know); will the names that are given when calling fp/FastifyPlugin be changed? Am I missing something?

I suppose it should change in next major, following with `fastify@4` release.

**Comment by RafaelGSS** [comment]
> @RafaelGSS the issue/thread you pointed at does not (at least to me) address what I am talking about. When you make a fastify plugin you give a plugin name (which probably know); will the names that are given when calling fp/FastifyPlugin be changed? Am I missing something?

oh, yeah. I assume it should be changed as well.

**Comment by jsumners** [comment]
Please read the comments above.

**Comment by BlackGlory** [comment]
Question: Why not use [npm deprecated](https://docs.npmjs.com/cli/v8/commands/npm-deprecate)?
Also, as far as I can tell, TypeScript types for deprecated packages are broken.

**Comment by jsumners** [comment]
npm depreciation notices are effectively useless. We may eventually add said notice to all of the packages as a polishing off detail, but it isn't a priority. Those notices are routinely ignored. The way depreciations were done here means they are much more difficult to ignore. 

In regard to TS stuff. If you read this full thread, you will see that I asked the community at large to help me determine if such things were accounted for properly. Unfortunately, that didn't happen and we needed to push forward because 1. we need to get Fastify v4 released and 2. I ran out of time for such a large endeavor.

**Comment by jsumners** [comment]
I have updated the original post with @RafaelGSS's list of remaining work.

**Comment by RafaelGSS** [comment]
@kibertoad can you move `fastify-schedule` to the fastify npm org? So we can bump a deprecated version

**Comment by kibertoad** [comment]
@RafaelGSS Will do! Btw, I tried to rename `fastify-awilix` to `@fastify/awilix`, but repo is refusing my push commits for some reason. Could you please check if I am still in the correct team?

**Comment by kibertoad** [comment]
@RafaelGSS I don't see an option on npmjs to transfer to org, and googling up npm command for that is not helping either. How can I transfer to org?

**Comment by jsumners** [comment]
> @RafaelGSS Will do! Btw, I tried to rename `fastify-awilix` to `@fastify/awilix`, but repo is refusing my push commits for some reason. Could you please check if I am still in the correct team?

You are in the `plugins` group and that group has "maintain" rights to that repo. You shouldn't be encountering any issues pushing. Have you tried re-cloning?

> @RafaelGSS I don't see an option on npmjs to transfer to org, and googling up npm command for that is not helping either. How can I transfer to org?

You have to give an org admin access (https://docs.npmjs.com/managing-team-access-to-organization-packages#adding-package-access-to-a-team-using-the-cli). I see that @mcollina is added as such, but if you add me then I'll get it done (`jsumners`).

**Comment by kibertoad** [comment]
@jsumners Invited you to both fastify-schedule and fastify-awilix

**Comment by jsumners** [comment]
> @jsumners Invited you to both fastify-schedule and fastify-awilix

I'll see about it tomorrow after work. Had a frustrating day today and will not be doing stuff tonight.

**Comment by jsumners** [comment]
> > You are in the plugins group and that group has "maintain" rights to that repo. You shouldn't be encountering any issues pushing. Have you tried re-cloning?
> 
> 
> 
> I have, and I even regenerated token, and it's still not working. My personal repos still work perfectly, so something fishy is going on here, I'll have to dig deeper.

Take it over to a team discussion. Post any error messages you are seeing.

**Comment by jsumners** [comment]
@fastify/plugins the `fastify-awilix` and `fastify-schedule` modules have been added to the npm org and you should have access to publish deprecation versions of them. You should also be able to publish the `@fastify/` versions of them, but reach out if not.

**Comment by RafaelGSS** [comment]
I think we can update the plugin docs explaining the latest version of each module is required Fastify v4

**Comment by jsumners** [comment]
> I think we can update the plugin docs explaining the latest version of each module is required Fastify v4

Not sure I understand. 

However, great job taking on the remaining modules.

**Comment by Fdawgs** [comment]
> > I think we can update the plugin docs explaining the latest version of each module is required Fastify v4
> 
> Not sure I understand.

I think what @RafaelGSS is saying is that the documentation for the plugins needs updating to state what plugin versions work with what Fastify versions, as we've had a few issues raised regarding it: 

- https://github.com/fastify/point-of-view/issues/320
- https://github.com/fastify/fastify-formbody/issues/114

**Comment by jsumners** [comment]
I have added https://github.com/fastify/any-schema-you-like to the list.

**Comment by mcollina** [comment]
What names should we pick for those non-fastify-branded packages?

**Comment by jsumners** [comment]
I don't have much of an opinion on that. I'd just prefix them with the org scope and move on.

**Comment by RafaelGSS** [comment]
I suggest:

- middie -> `@fastify/middie`

- point-of-view -> `@fastify/view`

- under-pressure -> `@fastify/under-pressure`

- any-schema-you-like -> `@fastify/any-schema-you-like`

**Comment by Fdawgs** [comment]
Maybe cut down any-schema-you-like to any-schema?

**Comment by jsumners** [comment]
Thank you for finishing the list @RafaelGSS. I feel bad that I got this thing started and then couldn't finish it. 

@fastify/plugins I have _removed_ all of the non-scoped packages from the `plugins` team on `npmjs.com`. They should not be getting any further updates, so the permissions should reflect that. Please raise an issue if you determine one needs a new version for some reason. We will coordinate with those who have permissions to do so.
